File: src/main/java/com/dg/watcher/history/History.java
package com.dg.watcher.history;

import com.dg.watcher.history.graph.HistoryGraph;
import hudson.model.AbstractProject;
import hudson.model.Action;
import org.kohsuke.stapler.StaplerRequest;
import org.kohsuke.stapler.StaplerResponse;
import java.io.IOException;


public class History implements Action {
    private AbstractProject<?, ?> project;
    private HistoryGraph graph;


    public History(AbstractProject<?, ?> project) {
        this.project = project;

        updateHistory();
    }

    public AbstractProject<?, ?> getProject() {
        return project;
    }

    @Override
    public String getIconFileName() {
        return "graph.gif";
    }

    @Override
    public String getDisplayName() {
        return "Apk History";
    }

    @Override
    public String getUrlName() {
        return "apk_history";
    }

    public void updateHistory() {
        graph = new HistoryGraph(project);
    }

    // Used in index.jelly
    public void doHistory(StaplerRequest request, StaplerResponse response) throws IOException {
        graph.drawGraph(request, response);
    }

    // Used in index.jelly
    public void doHistoryTooltips(StaplerRequest request, StaplerResponse response) throws IOException {
        graph.drawGraphTooltips(request, response);
    }
}

File: src/main/java/com/dg/watcher/history/History.kt
package com.dg.watcher.history

import com.dg.watcher.base.Project
import com.dg.watcher.history.graph.HistoryGraph
import hudson.model.Action
import org.kohsuke.stapler.StaplerRequest
import org.kohsuke.stapler.StaplerResponse


class History(val project: Project) : Action {
    private var graph: HistoryGraph? = null

    init {
        updateHistory()
    }

    override fun getIconFileName() = "graph.gif"

    override fun getDisplayName() = "Apk Size History"

    override fun getUrlName() = "apk_size_history"

    fun updateHistory() {
        graph = HistoryGraph(project)
    }

    // Used in index.jelly
    fun doHistory(request: StaplerRequest, response: StaplerResponse) {
        graph?.drawGraph(request, response)
    }

    // Used in index.jelly
    fun doHistoryTooltips(request: StaplerRequest, response: StaplerResponse) {
        graph?.drawGraphTooltips(request, response)
    }
}